### DISCUSSION TITLE
[Feature] 3D Stages

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#830] [Feature Request] [answered]
Several games have this where the stage is a 3D map rather than just a 2D sprites. (under night in birth is one I can recall) A potential implementation could be achieved under the BDdef statedef by pointing to a map file either in the .map or .vmf format (depends upon which one is more robust)

 >[BGdef]
>;Filename of sprite data
>map = e1m1.map`

along with the camrea info being slightly altered to accommodate the starting position

>[Camera]
>locationpos = 200,300,100 ;tells the camera where to start within the map file`

and the bounds to accommodate objects within the map upon being bounds

>[Bound]
>objectsAsBounds = 1; Uses objects that are within the map file as the bounds for the stage
>for instance a wall that the characters can't get past;

I'm sure there's more that could be done with this other than what my possible implementation has shown.

**Comment by Windblade-GR01** [comment]
We are using OpenGL so using Quake's map format maybe not the best.

A more standart format like OBJ would be better.
It does not support animation but it works better for a first implementation.

**Comment by Kikasuru** [comment]
This is bumping an old push request, but maybe we could use the glTF format?  It supports animations and is fairly easy to implement.

Hell, [there's already a library for it](https://github.com/qmuntal/gltf), though I'm not sure about the usability since I've never used it.  ...or GoLang for that matter, ahahah...

**Comment by K4thos** [comment, marked as answer]
already implemented
